src/libexpr/eval.hh

```diff
@@ -582,25 +582,41 @@ struct EvalSettings : Config
     Setting<Strings> nixPath{
         this, {}, "nix-path",
         R"(
-          List of directories to be searched for `<...>` file references.
+          List of directories to be searched for file references in `<...>` search path expressions.
+          It is overridden by the [`NIX_PATH`](@docroot@/command-ref/env-common.md#env-NIX_PATH)
+          environment variable, and can be extended using the [`-I` option](@docroot@/command-ref/opt-common.md#opt-I).
 
-          If [pure evaluation](#conf-pure-eval) is disabled,
-          this is initialised using the [`NIX_PATH`](@docroot@/command-ref/env-common.md#env-NIX_PATH)
-          environment variable, or, if it is unset and [restricted evaluation](#conf-restrict-eval)
-          is disabled, a default search path including the user's and `root`'s channels.
+          When nothing is specified:
```

Comment:
This implies that if something *is* specified, the `nix-path` setting is used when pure evaluation is enabled. Instead it should say that the resolution of `<...>` expressions is disabled in pure mode.